crates/snapbox/src/data.rs

```diff
@@ -24,6 +30,47 @@ impl Default for DataFormat {
     }
 }
 
+#[cfg(feature = "structured-data")]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    Number(serde_json::Number),
+    String(String),
+    Array(Vec<Value>),
+    // IndexMap is used as it keeps insertion order
+    Object(indexmap::IndexMap<String, Value>),
+}
```

Comment:
Why a custom `Value`?

Reply:
I believe we talked about having one type for all structured data but we didn't know what it would look like after a few formats. We decided to use `serde_json` until we need something different. Having the type be custom allows it to be extended and also allows for different types to be serialized and deserialized into it (`serde_json::from_str::<Value>(&data)`, `serde_json::to_string_pretty(&value)`, etc).

Reply:
I think there might be a disconnect as I find the following two statements contradictory:

> We decided to use serde_json

> Having the type be custom 

Could you also clarify the benefits?
- > Having the type be custom allows it to be extended 
  - In which ways are we wanting it extended?
- > also allows for different types to be serialized and deserialized into it
  - How is a custom type needed for this?

Reply:
I think I incorrectly thought that using `serde_json` meant basing the shared format on `serde_json`, which might be leading to my confusion. 

I think you are correct that we don't need a custom type currently. My idea about deserializing and serializing is wrong since you can specify the type to ser/de into. As for extensions, I mostly mean that if there is something a format in the future needs beyond `serde_json` having it be custom would allow an easy way to add that new variant.